### PR TITLE
शाश्वत परिक्रमं प्रक्षिणाति

### PR DESCRIPTION
**Pull request by performance**
Fixes issue #2. 

Simplifies the code to get player names.

### CODE REVIEW

File: 005-dy/ttti.js [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-  p2 = window.prompt(p1 === p2
-    ? `Please enter a different name than ${p1}.`
-    : 'द्वितीयस्य (२) नाम किम् - 2nd - dvitiiyasya naama kim ? [☺]')
-}
-
 window.onload = () => {
+  let p1 =  window.prompt('प्रथमस्य (१) नाम किम्? - 1st - prathamasya naama kim ? [x]', 'प्रथमः') 
+  p1 ??= 'प्रथमः'; // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment
+  let p2 = window.prompt(p1 === 'द्वितीयः'
+      ? `Please enter a different name than ${p1}.`
```
Comment (ezoerner) [inline 005-dy/ttti.js:5]: The logic here doesn't seem to do what is intended. This doesn't actually compare p1 with p2, so it's still possible to put in the same name for both players.

Comment (ezoerner) [reply 005-dy/ttti.js:5]: There still needs to be a loop if the same name was put in for p2  as p1 and only then should the prompt be changed to `Please enter a different name than ${p1}.`

**Review by ezoerner — CHANGES_REQUESTED** [review]

File: 005-dy/ttti.js (hunk 1 above)
Comment (ezoerner) [reply 005-dy/ttti.js:5]: I think the following code would work (and might be what you intended?):
```  let p1, p2
  p1 = window.prompt('प्रथमस्य (१) नाम किम्? - 1st - prathamasya naama kim ? [x]', 'प्रथमः')
  p1 ??= 'प्रथमः'

  while (!p2 || p1 == p2) {
    p2 = window.prompt((p1 === p2
      ? `Please enter a different name than ${p1}.`
      : 'द्वितीयस्य (२) नाम किम् - 2nd - dvitiiyasya naama kim ? [☺]'), 'द्वितीयः')
    p2 ??= (p1 === 'द्वितीयः') ? 'प्रथमः' : 'द्वितीयः';
  }
```

Comment (ezoerner) [reply 005-dy/ttti.js:5]: The Cancel button in the prompts here just set the names to the defaults, which is the same as hitting Ok to accept the defaults. This is probably acceptable, but ideally I would think Cancel should escape out of the app entirely and go back to the index page.

Comment (performance) [reply 005-dy/ttti.js:5]: You are right, this change does not handle the case where the user inputs a non default name for the first player and then enters the same thing for the second player.  Let me test your suggestions out and update the request. :)

Comment (ezoerner) [reply 005-dy/ttti.js:5]: > The Cancel button in the prompts here just sets the name to the default (which is the same as hitting Ok to accept the default). This is probably acceptable, but ideally I would think Cancel should escape out of the app entirely and go back to the index page.

… or even better, is there a way to put up a prompt window that doesn't have a Cancel button at all?

**Review by ezoerner — APPROVED** [review]

**Pull request merged**